Kids interface

noramyer opened the pull request.

noramyer committed:
Initial Framework for JavaFx GUI. Classes for different opjects begun, error in loading .fxml file.

noramyer committed:
Initial Framework for JavaFx GUI. Classes for different opjects begun, error in loading .fxml file.

noramyer committed:
Initial Framework for JavaFx GUI. Classes for different opjects begun, error in loading .fxml file.

noramyer committed:
-Created main objects for working chat and fixed bug for the application to run
-To Do: Finalize all button actions and get a working list of conversations

turtlefarmer commented on .idea/libraries/third_party.xml:
```diff
@@ -0,0 +1,12 @@
+<component name="libraryTable">
```
the .idea, bin, and *.iml files should not be in the repo.  The testing branch alraedy has a .gitignore file to exclude these.  You should remove them from your branch before merging.

turtlefarmer commented on run_simple_gui_client2.sh:
```diff
@@ -0,0 +1,21 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+LOCAL_MACHINE="localhost@2007"
```
Please modify this script to take the same inputs as the original run_client.sh.  Also, make sure you update the README for instructions to run the new UI as well.

turtlefarmer commented on src/codeu/chat/SimpleGuiClientMain2.java:
```diff
@@ -0,0 +1,100 @@
+      final Controller controller = new Controller(source);
+      final View view = new View(source);
+
+      LOG.info("Creating client...");
```
Some of the logging feels extraneous.  Generally there's not much use for logging which point the program has executed to, I assume a lot of this was used just for debugging?

turtlefarmer commented on src/codeu/chat/client/simplegui2/AlertBox.java:
```diff
@@ -0,0 +1,63 @@
+public final class AlertBox {
+
+    private Stage window;
+    private Label label1;
```
better variable name than label1

turtlefarmer commented on src/codeu/chat/client/simplegui2/AlertBox.java:
```diff
@@ -0,0 +1,63 @@
+    public AlertBox(String title, String message){
+
+        this.window = new Stage();
+        initialize(title, message);
```
I'm not a fan of the initialize functions if they are only ever called in the constructor.

turtlefarmer commented on src/codeu/chat/client/simplegui2/AlertBox.java:
```diff
@@ -0,0 +1,63 @@
+    private void initialize(String title, String message){
+
+        //creates window and sizing
+        window.setMinWidth(500);
```
If these are static, define them as class constants.
It might be a nice-to-have to dynamically size size this window based on screen resolution.

turtlefarmer commented on src/codeu/chat/client/simplegui2/AlertBox.java:
```diff
@@ -0,0 +1,63 @@
+        //inhibits user functions while window is open
+        window.initModality(Modality.APPLICATION_MODAL);
+
+        //Button 1
```
I think the code here is straight forward enough that you don't need these comments.  In general, it's better to make your code (variables, function names, function call flow, etc) to be self-describing so you don't have to explain what's happening in the comments.

turtlefarmer commented on src/codeu/chat/client/simplegui2/ChatMenu.java:
```diff
@@ -0,0 +1,200 @@
+
+        //add buttons for different options
+        final Button signOut = new Button("Sign Out");
+        signOut.setPrefSize(100, 20);
```
these button sizes should be stored in a shared variable.  Alternatively you could write a function that generates a button of your default size and other properties.

turtlefarmer commented on src/codeu/chat/client/simplegui2/ChatMenu.java:
```diff
@@ -0,0 +1,200 @@
+        addUser.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent actionEvent) {
+                NewUser user1=new NewUser(clientContext);
```
variable name.

turtlefarmer commented on src/codeu/chat/client/simplegui2/ChatMenu.java:
```diff
@@ -0,0 +1,200 @@
+
+                    if (s != null && s.length() > 0) {
+                        clientContext.conversation.startConversation(s, clientContext.user.getCurrent().id);
+                        //ConversationPanel.this.getAllConversations(listModel);
```
delete commented out code

turtlefarmer commented on src/codeu/chat/client/simplegui2/ChatMenu.java:
```diff
@@ -0,0 +1,200 @@
+        Label sideLabel=new Label("Conversations");
+
+        //border for box
+        sidebar.setStyle("-fx-padding: 10;" +
```
Consider:  Writing a class to encapsulate style key:value pair definitions.  Using strings like this can be very error prone.

turtlefarmer commented on src/codeu/chat/client/simplegui2/ChatMenu.java:
```diff
@@ -0,0 +1,200 @@
+            @Override
+            public void handle(ActionEvent actionEvent) {
+
+                //if contains bad word, alert box
```
haha i like this.  you might want to connect with @rshar1, perhaps instead of bad word detection you show an alert box if sentiment is *really really* negative for a single message.

turtlefarmer commented on src/codeu/chat/client/simplegui2/Controller.java:
```diff
@@ -0,0 +1,4 @@
+package codeu.chat.client.simplegui2;
```
Does this class do anything?

noramyer replied:
This one I wasn't sure if Javafx required a controller class to function....so I had left it in there but I will research and remove if necessary

turtlefarmer commented on src/codeu/chat/client/simplegui2/NewUser.java:
```diff
@@ -0,0 +1,92 @@
+
+        //Sign in button
+        Button signInButton=new Button("Sign in");
+        Button close= new Button("Exit.");
```
Buttons shouldn't have trailing periods.

noramyer committed:
Closed #16 problems I believe and updated the button actions

noramyer committed:
Implemented method to check for a bad word

rshar1 committed:
removed the .idea, bin, and *.iml from repo

rshar1 committed:
format changes and removed unnecessary statements

I auto formatted the code according to Google style guide. Also,
removed the fxml and controller since they were not being used.

rshar1 committed:
added gitignore

rshar1 committed:
corrected issue with shell

noramyer committed:
Merge pull request #32 from turtlefarmer/kids-interface-Raami

Preparation of merge for pull request

noramyer committed:
Merge branches 'kids_interface' and 'testing' of https://github.com/turtlefarmer/codeu_project_2017 into kids_interface

rshar1 committed:
Merge branch 'testing' into kid-interface-raami

rshar1 committed:
broadcast receiver added

The broadcast receiver has been added to update the messages but it
cannot join any conversations yet.

noramyer committed:
Merge pull request #35 from turtlefarmer/kid-interface-raami

Kid interface raami

noramyer committed:
Conversation changed to ListView

noramyer committed:
Merge branch 'kids_interface' of https://github.com/turtlefarmer/codeu_project_2017 into kids_interface

noramyer committed:
Conversation event listeners

noramyer committed:
Conversation selected item listener action

noramyer committed:
Added Index check

rshar1 committed:
receiver tracks changes in conversation

noramyer committed:
Merge pull request #36 from turtlefarmer/kids-interface-realTimeSupport

receiver tracks changes in conversation
